common/network/socket_instance.hpp

```diff
@@ -0,0 +1,20 @@
+/* -*-mode:c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#ifndef SOCKETINSTANCE_HPP
+#define SOCKETINSTANCE_HPP
+
+#include "address.hpp"
+#include "exception.hpp"
+#include "socket.hpp"
+
+/* class for network sockets (UDP, TCP, etc.) */
+class SocketInstance {
+public:
+    static UDPSocket socket;
```

Comment:
Is the default UDP?

Reply:
No, this is not the one actually used in offload_vio ([used here](https://github.com/ILLIXR/ILLIXR/blob/0329c6b5ad6b3a505e06c7a3ccdbd912fb0dae75/offload_vio/device_tx/plugin.cpp#L232)